src/mappings/v1/mint.ts

```diff
@@ -69,6 +70,11 @@ export async function mintItem(context: Context): Promise<void> {
       }
     }
 
+    const token = await handleTokenEntity(context, collection, final)
+    if (token) {
+      final.token = token
+    }
```

Comment:
handleTokenEntity can return `Promise<null | TokenEntity>` and you do not need if statement

Reply:
done